pkg/operator/controller/dns/controller.go

```diff
@@ -352,6 +361,24 @@ func (r *reconciler) publishRecordToZones(zones []configv1.DNSZone, record *iov1
 			}
 		} else if isRecordPublished {
 			condition, err = r.replacePublishedRecord(zones[i], record)
+		} else if isDomainPublished, err = domainIsAlreadyPublishedInZone(context.Background(), r.cache, record, &zones[i]); err != nil {
+			log.Error(err, "failed to validate DNS record", "record", record.Spec)
+			condition = iov1.DNSZoneCondition{
+				Message:            "Pre-publish validation failed",
+				Reason:             "InternalError",
+				Status:             string(operatorv1.ConditionUnknown),
+				Type:               iov1.DNSRecordPublishedConditionType,
+				LastTransitionTime: metav1.Now(),
+			}
+		} else if isDomainPublished {
+			log.Info("DNS record not published: domain name already used by another DNS record", "record", record.Spec)
+			condition = iov1.DNSZoneCondition{
+				Message:            "Domain name is already in use",
+				Reason:             "DomainAlreadyInUse",
+				Status:             string(operatorv1.ConditionUnknown),
+				Type:               iov1.DNSRecordPublishedConditionType,
+				LastTransitionTime: metav1.Now(),
```

Comment:
I believe `mergeConditions` will set `LastTransitionTime`, so you don't need to set it here (lines 371 and 380).

Reply:
https://github.com/openshift/cluster-ingress-operator/pull/1125/commits/523461a512528f13675de1204e3d6ad381974dbd updated `mergeConditions` to set `LastTransitionTime`.  That's a fairly recent change, so older code (e.g. line 360) might be setting `LastTransitionTime` because it used to be necessary but no longer is.